### PR TITLE
[SQL] Upgrade azure-mgmt-sql to 0.26.0

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

**Comment by Juliehzl** [comment]
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerMgmtScenarioTest::test_sql_server_public_network_access_update_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerDbMgmtScenarioTest::test_sql_db_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerDbMgmtScenarioTest::test_sql_db_read_replica_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerDbLongTermRetentionScenarioTest::test_sql_db_long_term_retention
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceOperationMgmtScenarioTest::test_sql_mi_operation_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerDbRestoreDeletedScenarioTest::test_sql_db_restore_deleted
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerDwMgmtScenarioTest::test_sql_dw_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerImportExportMgmtScenarioTest::test_sql_db_import_export_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceMgmtScenarioTest::test_sql_managed_instance_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceMgmtScenarioIdentityTest::test_sql_managed_instance_create_identity_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstancePoolScenarioTest::test_sql_instance_pool
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceDbShortTermRetentionScenarioTest::test_sql_managed_db_short_retention
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceDbLongTermRetentionScenarioTest::test_sql_managed_db_long_term_retention
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceRestoreDeletedDbScenarioTest::test_sql_managed_deleted_db_restore
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceDbMgmtScenarioTest::test_sql_managed_db_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlFailoverGroupMgmtScenarioTest::test_sql_failover_group_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlVirtualClusterMgmtScenarioTest::test_sql_virtual_cluster_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlInstanceFailoverGroupMgmtScenarioTest::test_sql_instance_failover_group_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlDbSensitivityClassificationsScenarioTest::test_sql_db_sensitivity_classifications
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedInstanceFailoverScenarionTest::test_sql_mi_failover_mgmt
FAILED src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlManagedDatabaseLogReplayScenarionTest::test_sql_midb_logreplay_mgmt

**Comment by Juliehzl** [comment]
Please comment in the PR with the following words and your PR link. For example,

Fix `test_sql_server_public_network_access_create_mgmt`: PR link

**Comment by emgu-ms** [comment]
Fix `test_sql_server_public_network_access_create_mgmt`: https://github.com/Azure/azure-cli/pull/16629

**Comment by Juliehzl** [comment]
> Fix `test_sql_server_public_network_access_create_mgmt`: #16629

Pass

**Comment by Oleza1972** [comment]
Fixed recorded tests:
test_sql_db_import_export_mgmt.yaml
test_sql_failover_group_mgmt.yaml
test_sql_instance_pool.yaml
test_sql_managed_db_mgmt.yaml
test_sql_managed_db_short_retention.yaml
test_sql_managed_deleted_db_restore.yaml
test_sql_managed_instance_create_identity_mgmt.yaml
test_sql_managed_instance_mgmt.yaml
test_sql_mi_failover_mgmt.yaml
test_sql_midb_logreplay_mgmt.yaml
test_sql_virtual_cluster_mgmt.yaml
test_sql_db_sensitivity_classifications.yaml

PR link: https://github.com/Azure/azure-cli/pull/16631

**Comment by Oleza1972** [comment]
Fixed additional tests:
test_sql_db_mgmt.yaml
test_sql_db_read_replica_mgmt.yaml
test_sql_dw_mgmt.yaml 

Was forced to change test_sql_commands.py  - 'readReplicaCount' and 'readScale' fields had incorrect value.
Probably 'readReplicaCount' field is no longer in user

PR link: https://github.com/Azure/azure-cli/pull/16631

**Comment by petrajko** [comment]
> * src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py::SqlServerImportExportMgmtScenarioTest::test_sql_db_import_export_mgmt

@nathannfan is probably owning this one

**Comment by Oleza1972** [comment]
Fixed tests to use 'highAvailabilityReplicaCount' field per @bradrich-msft request

**Comment by Oleza1972** [comment]
Fixed additional recording YAML files
PR link: https://github.com/Azure/azure-cli/pull/16631

**Comment by Oleza1972** [comment]
Fixed test_sql_db_import_export_mgmt
PR link: https://github.com/Azure/azure-cli/pull/16631

**Comment by Oleza1972** [comment]
@Juliehzl please ensure you merge commit:
https://github.com/Azure/azure-cli/pull/16631/commits/2d8a050a688359591ec1db492f4c675bf345dc78

**Comment by Oleza1972** [comment]
Fixed test_sql_db_restore.yaml
PR link: https://github.com/Azure/azure-cli/pull/16631

**Review by ericshape — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/sql/_params.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def _configure_db_dw_params(arg_ctx):
     arg_ctx.argument('read_scale',
                      arg_type=read_scale_param_type)
 
-    arg_ctx.argument('read_replica_count',
+    arg_ctx.argument('high_availability_replica_count',
                      arg_type=read_replicas_param_type)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:351]: Should Line537 be renamed too? https://github.com/Azure/azure-cli/blob/6ab0ebf4b419ca191ded239f6e93bdada86816d3/src/azure-cli/azure/cli/command_modules/sql/_params.py#L537-L539

**Review by evelyn-ys — APPROVED** [review]
Leave https://github.com/Azure/azure-cli/pull/16618#discussion_r565060661 for later check (another PR to fix if necessary).
Approved to unblock test fixes🙂

**Pull request merged**